### ISSUE TITLE
Support for UPDATE & DELETE methods?

### ISSUE THREAD

**Issue by gms8994** [cakephp/phinx#96]
Rob, have you given any thought to creating `insert`, `update`, and `delete` methods instead of having queries written inline?

I don't know that they'd add any specific functionality, but something like the way Zend_Db (v1, not v2) does would be nice to have.

**Comment by Keloo** [comment]
+1 for lastInsertId()

**Comment by trinhhunganh** [comment]
+1 for lastInsertId()

**Comment by cballou** [comment]
+1 for lastInsertId()

**Comment by atnartur** [comment]
+1 for lastInsertId()

**Comment by robmorgan** [comment]
I'm pushing this to 0.6.0 in order to get 0.5.2 out the door - it requires a a lot more planning and effort.

**Comment by WillGibson** [comment]
I'm just working on migrations to remove a whole load of redundant features from a legacy app before we start refactoring in earnest.

Would be great to have a delete() method in \Phinx\Db\Table.

**Comment by mbrodala** [comment]
I wonder how the API for an `update()` method could look like? Maybe like this:

```php
$table = $this->table('foo');
$row = ['id' => 1, 'title' => 'Bar'];
$matchFields = ['id'];
$table->update($row, $matchFields);
```

Which would result in the following SQL query:

```sql
UPDATE `foo` SET `title` = "Bar" WHERE `id` = 1
```

A similar API for `delete()` could also work.

**Comment by salmander** [comment]
Any updates on `lastInsertId` feature?

**Comment by arielallon** [comment]
+1 for lastInsertId. following.

**Comment by salmander** [comment]
My current work around:
```
SELECT 
  currval(pg_get_serial_sequence('{$this->table}', 'id')) AS id
```

**Comment by dellert** [comment]
+1 for lastInsertId()

**Comment by Assada** [comment]
+1 for lastInsertId()

**Comment by strategy** [comment]
+1 for lastInsertId()

**Comment by skaos** [comment]
+1 for lastInsertId()

**Comment by draobrehtom** [comment]
$con = $this->adapter->getAdapter()->getConnection();`
$con = $con->lastInsertId();

**Comment by WGPSenshi** [comment]
The above became necessary because adapters are now (0.9+) wrapped by adapterwrappers (usually TimedOutputAdapter), which do not have a getConnection() function.

For most cases,
$this->getAdapter()->getAdapter()->getConnection();
will suffice, but if you for some reasons have multiple wrappers, you'll need to iterate through them.

**Comment by josuamarcelc** [comment]
+1 for lastInsertId() + LOVE

**Comment by aegrey** [comment]
+1 for lastInsertId()

**Comment by lorenzo** [comment]
implemented in `v0.10.0`

**Comment by treadmillian** [comment]
Sorry, what was implemented? insert() and delete() methods or the lastInsertId()

**Comment by lorenzo** [comment]
@treadmillian look for queryBuilder I’m the documentation. Sorry I cannot be more specific, on mobile right now

**Comment by lorenzo** [comment]
Through the query builder you can do insert, update, delete and also get the lastInsertId

**Comment by treadmillian** [comment]
@lorenzo Thanks for that.  I noticed, however, that the documentation isn't complete, easy enough to understand to perform a `delete()` operation: http://docs.phinx.org/en/latest/migrations.html#using-the-query-builder

One thing I also noticed about the `queryBuilder()` there doesn't appear to be a way of specifying joins in a query, the examples only show straight selects on one table only.  Is this a WIP?

**Comment by lorenzo** [comment]
@treadmillian would you like to help improve the docs? I can guide you with some examples and the you can add them there :)

**Comment by lorenzo** [comment]
@treadmillian actually I just realised that the docs are not rendering some examples correctly. I'll go and fix that

**Comment by treadmillian** [comment]
@lorenzo Yep, that's much better, thank you.  Let me know how I go about updating the documentation.

**Comment by lorenzo** [comment]
@treadmillian that page should have an "edit in GitHub" button you can use to propose changes to the examples or explanations :)

**Comment by treadmillian** [comment]
@lorenzo Completely missed that link!  Thanks.
